pytest_splunk_addon/splunk.py

```diff
@@ -636,22 +898,77 @@ def splunk_external(request):
 
 
 @pytest.fixture(scope="session")
-def sc4s_docker(docker_services, tmp_path_factory, worker_id):
+def sc4s_kubernetes():
     """
     Provides IP of the sc4s server and related ports based on pytest-args(splunk_type)
     """
-    if worker_id:
-        # get the temp directory shared by all workers
-        root_tmp_dir = tmp_path_factory.getbasetemp().parent
-        fn = root_tmp_dir / "pytest_docker"
-        with FileLock(str(fn) + ".lock"):
-            docker_services.start("sc4s")
-
-    ports = {514: docker_services.port_for("sc4s", 514)}
+    if (
+        "PYTEST_XDIST_WORKER" not in os.environ
+        or os.environ.get("PYTEST_XDIST_WORKER") == "gw0"
+    ):
+        LOGGER.info(
+            "Starting kubernetes_service=sc4s for worker id {}".format(
+                str(os.environ.get("PYTEST_XDIST_WORKER"))
+            )
+        )
+        current_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "k8s_manifests",
+            "sc4s",
+        )
+        os.environ[
+            "SPLUNK_URL"
+        ] = "splunk-s1-standalone-service.{0}.svc.cluster.local".format(
+            os.getenv("NAMESPACE_NAME")
+        )
+        update_k8s_manifest_files(folder=current_path, file="sc4s_deployment")
+        LOGGER.info("Setting up SC4S")
+        kubernetes_helper_sc4s = KubernetesHelper()
+        kubernetes_helper_sc4s.create_k8s_resource_from_yaml(
+            file=os.path.join(current_path, "sc4s_deployment_updated.yaml"),
+            namespace_name=os.getenv("NAMESPACE_NAME"),
+        )
+        kubernetes_helper_sc4s.wait_for_kubernetes_resource_to_get_available(
+            "sc4s", os.getenv("NAMESPACE_NAME")
+        )
+        sc4s_pod_name = kubernetes_helper_sc4s.get_pod_name(
+            os.getenv("NAMESPACE_NAME"), "app=sc4s"
+        )
+        kubernetes_helper_sc4s.wait_for_pod_to_get_ready(
+            pod_name=sc4s_pod_name, namespace_name=os.getenv("NAMESPACE_NAME")
+        )
+        kubernetes_helper_sc4s.create_k8s_resource_from_yaml(
+            file=os.path.join(current_path, "sc4s_service.yaml"),
+            namespace_name=os.getenv("NAMESPACE_NAME"),
+        )
+        sc4s_ports_file = os.path.join(
+            os.getenv("TEST_RUNNER_DIRECTORY"), "exposed_sc4s_ports.log"
+        )
+        # Below kubectl port-forward can be replaced in future with kubernetes-client or any equivalent
+        sc4s_setup = subprocess.Popen(
+            "kubectl port-forward svc/sc4s-service -n {0} :514 > {1} 2>&1 &".format(
+                os.getenv("NAMESPACE_NAME"), sc4s_ports_file
+            ),
+            shell=True,
+        )
```

Comment:
Is it a big effort to replace this subprocess with kubernetes-client implementation?

Reply:
It will take extra efforts, we can't hold the whole task for this, it can be taken as a separate task, also created a JIRA for that, https://splunk.atlassian.net/browse/AQA-84